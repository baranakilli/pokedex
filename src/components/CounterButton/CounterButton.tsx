import { Component } from 'react';

interface CounterButtonState {
  count: number;
}

class CounterButton extends Component<{}, CounterButtonState> {
  constructor(props: {}) {
    super(props);
    this.state = {
      count: 0,
    };
  }

  shouldComponentUpdate(nextProps: {}, nextState: CounterButtonState) {
    return this.state.count !== nextState.count;
  }

  updateCount = () => {
    this.setState((state) => ({ count: state.count + 1 }));
  };

  render() {
    return (
      <button
        id="counter"
        className="mb3 ph3 pv1 br4 bw1"
        onClick={this.updateCount}
      >
        Count: {this.state.count}
      </button>
    );
  }
}

export default CounterButton;
