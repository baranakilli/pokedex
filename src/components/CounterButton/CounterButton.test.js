import { shallow } from 'enzyme';
import CounterButton from './CounterButton';

it('renders without crashing', () => {
  expect(shallow(<CounterButton/>)).toMatchSnapshot();
})

it('correctly increments the counter', () => {
  const wrapper = shallow(<CounterButton/>);
  wrapper.find('#counter').simulate('click')
  expect(wrapper.state().count).toEqual(1)
  wrapper.find('#counter').simulate('click')
  wrapper.find('#counter').simulate('click')
  expect(wrapper.state().count).toEqual(3);
  expect(wrapper).toMatchSnapshot();
});
