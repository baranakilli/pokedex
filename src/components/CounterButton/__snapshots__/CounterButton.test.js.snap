// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly increments the counter 1`] = `
<button
  className="mb3 ph3 pv1 br4 bw1"
  id="counter"
  onClick={[Function]}
>
  Count: 
  3
</button>
`;

exports[`renders without crashing 1`] = `
<button
  className="mb3 ph3 pv1 br4 bw1"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
